### PR TITLE
chore(rds): validation for master username with DatabaseInstanceFromSnapshot

### LINKED ISSUE #5002: rds.DatabaseInstanceFromSnapshot does not correctly set POSTGRES master user name
When using          

```
this.myrds = new rds.DatabaseInstanceFromSnapshot(this, `dbid`, {
engine: rds.DatabaseInstanceEngine.POSTGRES,
instanceClass: ec2.InstanceType.of(ec2.InstanceClass.T3, ec2.InstanceSize.MEDIUM),
snapshotIdentifier: 'rds:snapshotid' ,
masterUsername: `newmasteruser`, 
deletionProtection: false,
allocatedStorage: 150,
vpc: this.vpc,
masterUserPassword: this.newmasterpass,
})
```

The masterUserPassword _**is**_ applied to the new database, but the masterUsername is not.

Documentation states:

```
masterUsername?🔹
Type: string (optional, default: inherited from the snapshot)
The master user name.
```

So, if specified, the new username should be set.

### Reproduction Steps

As above.

### Error Log

The master user name is inherited from the snapshot, regardless of this setting.

### Environment

  - **CLI Version      : 1.15 **
  - **Framework Version: 1.15 **
  - **OS               : Linux **
  - **Language         : typescript **

### Other

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by jogold**
Better validation and improved doc for `masterUsername` in `DatabaseInstanceFromSnapshot`.

Closes #5002

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by skinny85** [comment]
This is almost impossible to review, because of the unrelated documentation changes that are included in the PR... ☹️

Is there any way to separate the doc changes from the logic changes?

**Comment by jogold** [comment]
> This is almost impossible to review

A bit exaggerated... anyway see #5167

**Comment by skinny85** [comment]
> > This is almost impossible to review
> 
> A bit exaggerated... anyway see #5167

I did say *almost* ^^

File: packages/@aws-cdk/aws-rds/lib/instance.ts [hunk 1]
```diff
@@ -822,8 +828,16 @@ export class DatabaseInstanceFromSnapshot extends DatabaseInstanceSource impleme
...
     }
 
+    if (!props.generateMasterUserPassword && props.masterUsername) {
+      throw new Error('Cannot specify `masterUsername` when `generateMasterUserPassword` is set to false.');
+    }
+
+    if (props.generateMasterUserPassword && props.masterUserPassword) {
+      throw new Error('Cannot specify `masterUserPassword` when `generateMasterUserPassword` is set to true.');
+    }
+
```
Comment (skinny85) [inline packages/@aws-cdk/aws-rds/lib/instance.ts]: I wonder if grouping these wouldn't make it a little easier to follow:

```ts
    let secret: DatabaseSecret | undefined;

    if (props.generateMasterUserPassword) {
      if (!props.masterUsername) {
        throw new Error('`masterUsername` must be specified when `generateMasterUserPassword` is set to true.');
      }

      if (props.masterUserPassword) {
        throw new Error('Cannot specify `masterUserPassword` when `generateMasterUserPassword` is set to true.');
      }
     
      // here you now `masterUsername` is defined, so no need to repeat the condition below
      secret = new DatabaseSecret(this, 'Secret', {
        username: props.masterUsername,
        encryptionKey: props.secretKmsKey,
      });
    } else {
      // props.generateMasterUserPassword is false
      if (props.masterUsername) {
        throw new Error('Cannot specify `masterUsername` when `generateMasterUserPassword` is set to false.');
      }
    }
```

**Pull request merged**